plugins/examples/custom-processor/src/yamlRestTest/resources/rest-api-spec/test/resthandler/20_process_document.yml

```diff
@@ -0,0 +1,65 @@
+"Process document":
+  # create ingest pipeline with custom processor
+  - do:
+      ingest.put_pipeline:
+        id: pipeline1
+        body:  >
+          {
+            "processors": [
+              {
+                "repeat" : {
+                  "field": "to_repeat"
+                }
+              }
+            ]
+          }
+  - match: { acknowledged: true }
+
+  # index a document with field to be processed
+  - do:
+      index:
+        id: doc1
+        index: index1
+        pipeline: pipeline1
+        body: { to_repeat: "foo" }
+  - match: { result: "created" }
+
+  # validate document is processed
+  - do:
+      get:
+        index: index1
+        id: doc1
+  - match: { _source: { to_repeat: "foofoo" } }
+---
+"Does not process document without field":
+  # create ingest pipeline with custom processor
+  - do:
+      ingest.put_pipeline:
```

Comment:
I'd recommend putting this in a `setup:` block so that you don't have to repeat yourself. Also you should probably remove it in a `teardown:`.

Reply:
`230_change_target_index.yml` has an example of those two blocks in action.